sdk/src/types/block/output/account.rs

```diff
@@ -388,6 +388,12 @@ impl AccountOutput {
         self.features.block_issuer().is_some()
     }
 
+    /// Returns whether the output can claim rewards based on its current and next state in a transaction.
+    pub fn can_claim_rewards(&self, next_state: Option<&Self>) -> bool {
+        // TODO: should destroying an account add rewards?
+        self.features().staking().is_some() && next_state.map_or(false, |o| o.features().staking().is_none())
```

Comment:
What if it's 2 different accounts ?

Reply:
Or we just expect this to be called with the right input/output?

Reply:
yeah that

Reply:
Ok fair enough